Libraries/libcar/Headers/car/Writer.h

```diff
@@ -60,6 +67,17 @@ class Writer {
      */
     void addRendition(Rendition const &rendition);
 
+    /*
+     * Add a rendition for a facet, optimized for fast editing of CAR files
+     */
+    void addRendition(void *key, size_t key_len, void *value, size_t value_len);
+
+    /*
+     * The key format, optional and determined automatically if omitted.
+     */
+    ext::optional<struct car_key_format *> &keyfmt()
+    { return _keyfmt; }
```

Comment:
I still think this isn't quite right. If you specify a key format here, it has to manually account for all attributes in non-raw renditions! If you don't, then they will silently be written out wrong.

Are you sure that computing a key format automatically is actually slow? Even for thousands of renditions, it shouldn't be much more expensive than just iterating over the renditions, since there are only a few possible attribute identifiers.